go/exorelay/features/receiving.feature

```diff
@@ -0,0 +1,46 @@
+Feature: Receiving messages
+
+  As an Exosphere developer
+  I want my code base to be able to respond to incoming messages
+  So that I can create Exoservices.
+
+  Rules:
+  - you need to register handlers for messages that you want to receive
+  - handlers are called with the request data
+
+
+  Background:
+    Given an ExoRelay with the role "test-service"
+    And ExoRelay connects to Exocom
+
+
+  Scenario: receiving a message without payload
+    Given I register a handler for the "hello-world" message
+    When receiving this message:
+      """
+      {
+        "name": "hello-world",
+        "id": "123"
+      }
+      """
+    Then ExoRelay runs the registered handler passing nil as the first argument
+
+
+  Scenario: receiving a message with Hash payload
+    Given I register a handler for the "hello" message
+    When receiving this message:
+      """
+      {
+        "name": "hello",
+        "payload": {
+          "name": "world"
+        },
+        "id": "123"
+      }
+      """
+    Then ExoRelay runs the registered handler, passing the following as the first argument
```

Comment:
We have the convention that if a step has an associated text block or table, it has a trailing `:`.

```cucumber
Then ExoRelay runs the registered handler with the argument: